src/basic/os-util.c

```diff
@@ -123,33 +123,35 @@ int open_extension_release(const char *root, const char *extension, char **ret_p
                                         continue;
                                 }
 
-                                /* No xattr or cannot parse it? Then skip this. */
-                                _cleanup_free_ char *extension_release_xattr = NULL;
-                                k = fgetxattr_malloc(extension_release_fd, "user.extension-release.strict", &extension_release_xattr);
-                                if (k < 0 && !ERRNO_IS_NOT_SUPPORTED(k) && k != -ENODATA)
-                                        log_debug_errno(k,
-                                                        "%s/%s: Failed to read 'user.extension-release.strict' extended attribute from file: %m",
+                                if (!extension_release_nonstrict) {
+                                        /* No xattr or cannot parse it? Then skip this. */
+                                        _cleanup_free_ char *extension_release_xattr = NULL;
+                                        k = fgetxattr_malloc(extension_release_fd, "user.extension-release.strict", &extension_release_xattr);
+                                        if (k < 0 && !ERRNO_IS_NOT_SUPPORTED(k) && k != -ENODATA)
+                                                log_debug_errno(k,
+                                                                "%s/%s: Failed to read 'user.extension-release.strict' extended attribute from file: %m",
+                                                                extension_release_dir_path, de->d_name);
+                                        if (k < 0) {
+                                                log_debug("%s/%s does not have user.extension-release.strict xattr, ignoring.", extension_release_dir_path, de->d_name);
+                                                continue;
+                                        }
```

Comment:
something is off here, as this will log twice about the same error in some cases.

Reply:
pre-existing condition, but will fix